test/framework/prometheus.go

```diff
@@ -398,10 +415,15 @@ func (f *Framework) WaitForPrometheusReady(ctx context.Context, p *monitoringv1.
 		},
 		timeout,
 	); err != nil {
-		return fmt.Errorf("prometheus %v/%v failed to become available: %w", p.Namespace, p.Name, err)
+		return nil, fmt.Errorf("prometheus %v/%v failed to become available: %w", p.Namespace, p.Name, err)
 	}
 
-	return nil
+	p, err := f.MonClientV1.Prometheuses(p.Namespace).Get(ctx, p.Name, metav1.GetOptions{})
```

Comment:
we don't need the extra Get() if we use the `current` variable.

Reply:
good find!